### PR TITLE
Document gatewayOncePerNamespace Application Operator work mode

### PR DESCRIPTION
**Pull request by majakurcius**
**Description**

Application Operator can now work in two modes. The default old one, where Application Gateway is deployed for each Application, and the newly implemented alternative mode, in which the Application Gateway is deployed once per Namespace based on the ServiceInstance custom resource. 

Changes proposed in this pull request:

- Document implemented feature flag to deploy Application Gateways once per Namespace.

**Related issue(s)**
See #7043, #7439

### CODE REVIEW

File: components/application-operator/README.md [hunk 1]
```diff
@@ -2,16 +2,25 @@
...
  - Application updated - the AO updates the Status of the Application Helm Release.
  - Application deleted - the AO deletes Helm chart corresponding to the given Application.
+ 
+<!--- when gatewayOncePerNamespace=true -->
+In the Gateway-per-Namespace mode:
+ - First ServiceInstance created in a given Namespace - the AO installs the Helm chart that contains all the necessary Kubernetes resources required for the Application Gateway to work.
+ - Last ServiceInstance from a given Namespace is deleted - the AO deletes the Gateway Helm chart.
 
 
 ## Usage
```
Comment (klaudiagrz) [inline components/application-operator/README.md]: In the opening line below, you start with `The`, whereas in other sections you start straight with AO. I'd get rid of `The`.

File: components/application-operator/README.md [hunk 2]
```diff
@@ -39,4 +49,4 @@ To test the component you modified, run the `run-with-local-tests.sh` script loc
 
 Running the script builds the Docker image of the component, pushes it to the Minikube registry, and updates the component deployment in the Minikube cluster. It then triggers the `run-local-tests.sh` script, which builds the image of the acceptance tests to the Minikube registry, creates a Pod with the tests, and fetches the logs from that Pod.
 
-Alternatively, you can run only the `run-local-tests.sh` script for the given component to build the image of the component's acceptance tests to the Minikube registry, create a Pod with the tests, and fetch the logs from that Pod.
+Alternatively, you can run only the `run-local-tests.sh` script for the given component to build the image of the component's acceptance tests to the Minikube registry, create a Pod with the tests, and fetch the logs from that Pod.
```
Comment (klaudiagrz) [inline components/application-operator/README.md]:
```suggestion
Alternatively, you can run only the `run-local-tests.sh` script for the given component to build the image of the component's acceptance tests in the Minikube registry, create a Pod with the tests, and fetch the logs from that Pod.
```

File: docs/application-connector/02-01-application-connector.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ The AB implements the [Open Service Broker API](https://www.openservicebrokerapi
 
 ## Application Operator
 
-The operator listens for creating or deleting the Application custom resources and acts accordingly, either provisioning or de-provisioning an instance of the Application Gateway and the Event Service for every custom resource.
+Application Operator can work in two modes. By default, the operator listens for creating or deleting the Application custom resources and acts accordingly, either provisioning or deprovisioning an instance of the Application Gateway and the Event Service for every custom resource. In the alternative mode, it listens for an additional resource, ServiceInstance. In this mode, it provisions an instance of the Application Gateway once per Namespace. That means that there is always only one Application Gateway per Namespace, even if there are more ServiceInstances and Applications. The Application Gateway gets deleted with the last ServiceInstance in that Namespace. The alternative mode is enabled by setting the `gatewayOncePerNamespace` feature flag to true.
```
Comment (klaudiagrz) [inline docs/application-connector/02-01-application-connector.md]: Same as above: do we need a link to the Application CR?

File: components/application-operator/README.md [hunk 4]
```diff
@@ -2,16 +2,25 @@
 
 ## Overview
 
-Application Operator detects changes in Application custom resources and acts accordingly.
+Application Operator can work in two modes. 
+By default, it detects changes in Application custom resources and acts accordingly. In this mode, Application Gateway is created for each Application.
+In the alternative mode, it detects changes in ServiceInstance custom resources and acts accordingly. In this mode, Application Gateway is created per Namespace.
 
 
 ## Performed operations
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply components/application-operator/README.md]: It's meant to be another section, not a subsection.

File: components/application-operator/README.md (hunk 1 above)
Comment (majakurcius) [reply components/application-operator/README.md]: I'll stick to `the` for consistency's sake, as all the other components use it, too, but I'll unify it.  Good catch!

**Pull request merged**